### PR TITLE
Issue 26003 implement sync feature for sites default site fixes

### LINKED ISSUE #26003: Implement "sync" feature for sites

### PR DESCRIPTION
**Pull request by jgambarios**
The purpose of this PR is to correctly handle changes to the `default` site property found in each site descriptor, thereby allowing the default site to be updated in accordance with that flag.

### CODE REVIEW

File: tools/dotcms-cli/cli/src/test/java/com/dotcms/cli/command/site/SiteCommandIT.java [hunk 1]
```diff
@@ -30,6 +30,7 @@
 import javax.inject.Inject;
 import javax.ws.rs.NotFoundException;
 import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jetbrains.annotations.NotNull;
```
Comment (fabrizzio-dotCMS) [inline tools/dotcms-cli/cli/src/test/java/com/dotcms/cli/command/site/SiteCommandIT.java:33]: I think this should go away we don't do jetbrains

**Review by fabrizzio-dotCMS — COMMENTED** [review]
just a few minor observations

File: tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/push/ContentComparator.java [hunk 2]
```diff
@@ -46,5 +47,31 @@ public interface ContentComparator<T> {
...
+        return new DefaultOrderComparator<>();
+    }
+
+    /**
+     * Default implementation of the ContentComparator interface that does not change the processing
+     * order.
+     *
+     * @param <T> the type of content to be compared
+     */
+    class DefaultOrderComparator<T> implements Comparator<T> {
```
[the remark this answers is not in the document]
Comment (jgambarios) [reply tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/push/ContentComparator.java]: Done

File: tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/model/push/AbstractPushAnalysisResult.java [hunk 3]
```diff
@@ -17,6 +17,8 @@ public interface AbstractPushAnalysisResult<T> {
 
     Optional<T> serverContent();
 
+    Optional<T> localContent();
+
     Optional<File> localFile();
```
[the remark this answers is not in the document]
Comment (jgambarios) [reply tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/model/push/AbstractPushAnalysisResult.java:51]: Done

File: tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/push/site/SiteComparator.java [hunk 4]
```diff
@@ -72,6 +73,15 @@ public boolean contentEquals(SiteView localSite, SiteView serverContent) {
         return localSite.equals(byId.entity());
     }
 
+    @Override
+    public Comparator<SiteView> getProcessingOrderComparator() {
+        return (site1, site2) ->
```
[the remark this answers is not in the document]
Comment (jgambarios) [reply tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/push/site/SiteComparator.java:92]: Done

File: tools/dotcms-cli/cli/src/test/java/com/dotcms/cli/command/site/SiteCommandIT.java (hunk 1 above)
Comment (jgambarios) [reply tools/dotcms-cli/cli/src/test/java/com/dotcms/cli/command/site/SiteCommandIT.java]: Done

**Review by fabrizzio-dotCMS — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Pull request merged**